lib/src/extensions/canvas_extension.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:ui';
+
+import 'package:path_drawing/path_drawing.dart';
+
+extension DashedPath on Canvas {
+  /// helper that generates dashed paths for other methods
+  void drawDashedPath(Path path, Paint painter, List<int> dashArray) {
```

Comment:
In my opinion, it's better to have two functions,
one to convert the path to dashed path,
and the other one to draw the path(we already have this function)
so please make an extension function on the Path, and as I said before call it `toDashedPath()`, and just draw the result as a normal path (using `canvas.drawPath()` function)
My consern is applying SRP (functions should do one and only one thing), and we can reuse the `toDashedPath()` later.

Reply:
It would be 3x the code to put it on path... I would need to call the generate function everywhere and do logic throughout the app

Reply:
234d35e

Reply:
I know your concern, and I know we should write more codes for it,
but It helps us to maintain the code easier.